### PR TITLE
Use placeholders for URLs in translation strings

### PR DESCRIPTION
**Pull request by leofeyer**
So that we can change the URLs without creating unnecessary work for the translators. See https://github.com/contao/to.contao.org/pull/80 for more information.

### CODE REVIEW

File: core-bundle/src/Resources/views/Error/invalid_request_token.html.twig [hunk 1]
```diff
@@ -11,5 +11,5 @@
 {% endblock %}
 {% block explain %}
     <p>{{ 'XPT.tokenExplainOne'|trans|raw }}</p>
-    <p>{{ 'XPT.tokenExplainTwo'|trans|raw }}</p>
+    <p>{{ 'XPT.tokenExplainTwo'|trans|format('https://to.contao.org/support')|raw }}</p>
```
Comment (Toflar) [inline core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: Shouldn't we be able to pass those as arguments to `trans()`?

Comment (leofeyer) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: Doesn‘t work for `sprintf()` placeholders. See e.g. https://github.com/contao/contao/blob/4.13/installation-bundle/src/Resources/views/configuration_error.html.twig.

Comment (Toflar) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: I thought it works for us? That's why we've decorated the translator? https://github.com/contao/contao/blob/5.x/core-bundle/src/Controller/BackendPreviewSwitchController.php#L134

Comment (leofeyer) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: Yes, but these are Twig templates and not Contao templates. 😉

Comment (Toflar) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: And? The `trans` filter uses the Symfony translator which we decorated. I've just tested it and it works perfectly fine:

`{{ 'ERR.maxval'|trans(['field', 2], 'contao_default') }}` outputs ` Field "field" must not be greater than 2! ` 🤷‍♂️

Comment (Toflar) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: Imho `{{ 'XPT.tokenExplainTwo'|trans(['https://to.contao.org/support'])|raw }}` should work too 😊

Comment (leofeyer) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: Ah, I didn‘t try an array argument. Hang on…

Comment (leofeyer) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: BTW, why did we not do this in the other templates (see above) as well?

Comment (Toflar) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: I don't know, this is the first time I see the `|format` filter :)

Comment (leofeyer) [reply core-bundle/src/Resources/views/Error/invalid_request_token.html.twig]: Changed in cfe309d0993e5da4635c2c26ee7544545a88cb58.

**Comment by leofeyer** [comment]
On second thought, passing the replacement strings to the `trans()` method ties the template to our translator. This way, the template is no longer compatible with the default Twig parser, which it would be if we continued to use `format()`.

@contao/developers What do you think? Do we care?

**Comment by Toflar** [comment]
I don't understand the issue. You want to translate a string, this always ties it to a translator? And what do you mean by "default Twig parser"? `trans` is a filter added by the [fullstack framework of Symfony (or the Bridge more specifically)](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bridge/Twig/Extension/TranslationExtension.php#L71) so it's already tied to the framework and it's not a Twig standalone filter at all.
Or in other words, if that were an issue, we were also not allowed to decorate the translator and use it the way we are already using it in many places (like [here for example](https://github.com/contao/contao/blob/5.x/core-bundle/src/Controller/BackendPreviewSwitchController.php#L134)).

**Comment by fritzmg** [comment]
The template is part of `contao/core-bundle` and `contao/core-bundle` decorates the `Translator` - thus there should never be an issue.

**Comment by leofeyer** [comment]
Ok, then I will merge this as soon as it has been approved.

**Review by Toflar — APPROVED** [review]
LGTM but I think we should still ask @m-vo for a review just because it's affecting Twig and I don't want to interfere with his plans here.

**Comment by leofeyer** [comment]
Agreed, although none of these templates are related to his work. This is Contao 4.13 after all. 😄

**Comment by m-vo** [comment]
Changes look good to me.

There is one thing we could think about, though: With HTML in the translated strings, we need to bypass the escaper (`|raw`). But ideally, we would encode anything but the link, which we could totally do now in Contao 5 with insert tags and the `insert_tag_raw` filter:

```xlf
<trans-unit id="XPT.tokenExplainTwo">
    <source>Learn more about speeding up your workflow by using {{link_open::%s::blank}}keyboard shortcuts{{link_close}}.</source>
</trans-unit>
```

```twig
<p>{{ 'XPT.tokenExplainTwo'|trans(['https://to.contao.org/support'])|insert_tag_raw }}</p>
```

Wdyt?

**Comment by ausi** [comment]
I don’t like the idea of using insert tags, especially things like `{{link_open::*}}` are problematic IMO.

I think the “correct” solution would be that the link comes from the templates like this (pseudocode):

```xlf
<trans-unit id="message">
    <source>Learn more about speeding up your workflow by using [keyboard shortcuts].</source>
</trans-unit>
```

```twig
{{ 'message'|trans({'[' => '<a href="…">', ']' => '</a>'}) }}
```

Not sure if this can be implemented in a way so that the message still gets encoded correctly.

But for now I think we can keep it as it is.

**Pull request merged**